libp2p/__init__.py

```diff
@@ -43,28 +33,27 @@ def generate_peer_id_from(key_pair: KeyPair) -> ID:
 
 
 def initialize_default_swarm(
-    key_pair: KeyPair,
-    id_opt: ID = None,
-    transport_opt: Sequence[str] = None,
+    key_pair: KeyPair = None,
     muxer_opt: TMuxerOptions = None,
     sec_opt: TSecurityOptions = None,
     peerstore_opt: IPeerStore = None,
-) -> Swarm:
+) -> INetworkService:
```

Comment:
Why are we changing to a more abstract type to the function output here?

Reply:
I intended to use interfaces over implementations. I can change it back if you think it's more reasonable.

Reply:
Using interfaces sounds good 👍

Reply:
Agreed.  This has helped us a lot in both the trinity and py-evm codebases to establish concrete interface definitions and to use them ubiquitously in our type hints.